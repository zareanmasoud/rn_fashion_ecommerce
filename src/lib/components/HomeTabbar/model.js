// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
