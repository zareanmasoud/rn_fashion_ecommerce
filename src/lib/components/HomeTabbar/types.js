// @flow

export type Buttons = {
  handlePress: Function,
  src: string,
  text: string,
};
