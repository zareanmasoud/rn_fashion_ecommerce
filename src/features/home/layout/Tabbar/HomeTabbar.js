import React from 'react';
import HomeTabbarComponent from 'lib/components/HomeTabbar';
import model from './model';

const HomeTabbar = () => {
  return <HomeTabbarComponent buttons={model.buttons} />;
};

export default HomeTabbar;
