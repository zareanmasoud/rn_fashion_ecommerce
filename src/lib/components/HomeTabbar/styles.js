import {StyleSheet} from 'react-native';
// eslint-disable-next-line import/no-extraneous-dependencies
import R from 'res/R';
// eslint-disable-next-line import/no-extraneous-dependencies
import {scale} from 'lib/utils/scaling';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: R.colors.whiteBackground,
    height: scale(50),
    ...R.palette.shadow('narrow'),
  },
  emptyView: {
    width: scale(20),
    height: scale(20),
  },
});
