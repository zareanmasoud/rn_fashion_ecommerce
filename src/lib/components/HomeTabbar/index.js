import HomeTabbar from './HomeTabbar';

export {HomeTabbar as default};
