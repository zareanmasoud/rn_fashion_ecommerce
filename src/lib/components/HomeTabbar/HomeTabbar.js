import React from 'react';
import {View} from 'react-native';
import TabbarButton from 'lib/components/TabbarButton';
import styles from './styles';
import type {Buttons} from './model';

type Props = {
  buttons: Buttons[],
};

const HomeTabbar = ({buttons}: Props) => {
  const TabbarButtons = Object.values(buttons).map(button => (
    <TabbarButton {...button} key={button.id} />
  ));

  return <View style={styles.container}>{TabbarButtons}</View>;
};

export default HomeTabbar;
