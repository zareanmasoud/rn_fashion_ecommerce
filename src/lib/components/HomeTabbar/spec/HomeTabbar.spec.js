import React from 'react';
import {shallow} from 'enzyme';
import input from './input';
import HomeTabbar from '../HomeTabbar';

describe('HomeTabbar Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<HomeTabbar buttons={input.buttons} />);
      expect(component).toMatchSnapshot();
    });
  });
});
