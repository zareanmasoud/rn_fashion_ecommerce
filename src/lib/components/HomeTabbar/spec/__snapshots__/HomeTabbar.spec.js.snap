// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTabbar Component Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "elevation": 1,
      "flexDirection": "row",
      "height": 107.14285714285714,
      "justifyContent": "space-between",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 0,
    }
  }
>
  <TabbarButton
    id={0}
    key="0"
    press={[Function]}
    src={
      Object {
        "process": [Function],
      }
    }
    text="Home"
  />
  <TabbarButton
    id={1}
    key="1"
    press={[Function]}
    src={
      Object {
        "process": [Function],
      }
    }
    text="Wishlist"
  />
  <TabbarButton
    id={2}
    key="2"
    press={[Function]}
    src={
      Object {
        "process": [Function],
      }
    }
    text="Help"
  />
  <TabbarButton
    id={3}
    key="3"
    press={[Function]}
    src={
      Object {
        "process": [Function],
      }
    }
    text="Profile"
  />
</View>
`;
